### PR TITLE
Use Puma as web server

### PR DESCRIPTION
**Pull request by sophiebrown01**
This PR overrides app `Dockerfile`s to use Puma as the default web server instead of Unicorn and will entirely sidestep the use of Foreman. Foreman isn't necessary when running on Kubernetes, as the controller managers within Kubernetes will take care of ensuring that our app is running without the need of a separate process manager.

Puma is started [as recommended in the docs](https://github.com/puma/puma#rails) by running `bundle exec puma` directly (rather than via `bundle exec rails` which would then use Puma as the default web server).

Co-authored-by: Chris Banks <chris.banks@digital.cabinet-office.gov.uk>

Trello: https://trello.com/c/irrbEFMV

### CODE REVIEW

File: charts/frontend/templates/deployment.yaml [hunk 1]
```diff
@@ -20,6 +20,8 @@ spec:
           ports:
             - name: http
               containerPort: {{ .Values.appPort }}
+          command: ["bundle"]
+          args: ["exec", "rails", "server"]
```
Comment (kerin) [inline charts/frontend/templates/deployment.yaml:24]: Should we not specify the port here to match the `containerPort` above?

```
["exec", "rails", "server", "-p", "{{ .Values.appPort }}"]
```

or such?

Comment (sengi) [reply charts/frontend/templates/deployment.yaml:24]: Good point but we're already setting it (perhaps in a slightly less clear way) by the `PORT` env var below. In principle we could remove `PORT` and use `-p`, but in practice that'd mean we'd have to change the Dockerfile in a way that's not compatible with its existing usage (in alphagov/publishing-e2e-tests).

Comment (kerin) [reply charts/frontend/templates/deployment.yaml:24]: Ah, forgot about that env var - no need to change that to an arg 👍

Comment (bilbof) [reply charts/frontend/templates/deployment.yaml:24]: I think the preferred way to set the port would be to set the environment variable (https://github.com/alphagov/frontend/blob/main/config/puma.rb#L18) which is read by the config file.

File: charts/publisher/templates/worker/deployment.yaml [hunk 2]
```diff
@@ -25,8 +25,8 @@ spec:
       containers:
         - name: app
           image: "{{ .Values.common.image.repository }}:{{ .Values.common.image.tag }}"
-          command: ["foreman"]
-          args: ["run", "worker"]
+          command: ["bundle"]
+          args: ["exec", "sidekiq", "-C", "./config/sidekiq.yml"]
```
Comment (sengi) [inline charts/publisher/templates/worker/deployment.yaml:29]: nit: the `./` is superfluous here
```suggestion
          args: ["exec", "sidekiq", "-C", "config/sidekiq.yml"]
```

File: charts/static/templates/deployment.yaml [hunk 3]
```diff
@@ -38,7 +40,7 @@ spec:
             - name: PORT
               value: "{{ .Values.appPort }}"
             - name: RAILS_ENV
-              value: "{{ .Values.govukEnvironment }}"
+              value: "{{ .Values.appRailsEnv }}"
```
Comment (sengi) [inline charts/static/templates/deployment.yaml:43]: Good catch!

**Review by sengi — APPROVED** [review]

**Pull request merged**